### DISCUSSION TITLE
PIXI.BitmapFont.from doesn't support Cyrilic and CJK languages.

### DISCUSSION THREAD

**Discussion by EvanetaI** [pixijs/pixijs#8663] [Q&A]
I'm using Roboto font to generate bitmap font with the help of PIXI.BitmapFont.from(). Then I'm creating text out of the generated font if the charset is Latin all good but with Cyrillic and CJK the text is not showing off the screen. The chars object which PIXI.BitmapFont.from() accepts has limited options. Roboto fonts support many languages, is it possible to use that or somehow set it when we create the bitmap font? I need this for a multilingual project. 
Here's the code: 

   PIXI.BitmapFont.from("RobotoFont", {
      fontFamily: "Roboto",
      fontSize: 100,
      strokeThickness: 2,
      fill: "purple",
    }//,{
      //chars:   // I tried all of the options nothing worked
    //}
    )

const title = new PIXI.BitmapText("男の子", { fontName: "RobotoFont" });

**Comment by bigtimebuddy** [comment]
This works for me:
https://jsfiddle.net/bigtimebuddy/cx2wf86m/

Perhaps your `chars` ranges don't work? You can alternately provide a string of characters to support.

**Comment by EvanetaI** [comment]
Thank you for your answer. Yes, this way is working. It will be overkilling if I have to list all chars needed. The application will support many languages. Is there any other way to include all supported languages charsets from Roboto font in a bitmap font generated with PIXI.BitmapFont.from()? If not, I accept any good ideas. I was thinking to do it the old way by using a tool to generate a texture for me and then using it with PIXI.BitmapText.

**Reply by bigtimebuddy** [reply]
Here's an article I wrote about BitmapText: https://medium.com/p/244b9f95f830

In general, glyph-heavy languages (CJK) should likely be using PIXI.Text not PIXI.BitmapText for memory considerations. Generating or pre-generating the atlas with all the glyphs is going to be very expensive to load. PIXI.Text is fine if you are not updating it constantly (static).

**Reply by EvanetaI** [reply]
I tried to pre-generate a bitmap font using Hiero, but the image with all glyphs itself is 340kb. And I have 4 fonts used in the project. So, if I have to pre-generate a bitmap atlas for each of them it will be too heavy to load.  I jumped to PIXI.BitmapText because I was avoiding using PIXI.Text, exactly for memory considerations. I tried signed-distance-filed for PIXI but the library was not compatible for some reason. The text in the application is updated only on resize or on language change, this is why I was thinking that I have to totally stop using PIXI.Text, because these operations might be too heavy.

**Reply by goldenratio** [reply]
PIXI.Text is fine in this scenario. We have shipped our game help section with PIXI.Text, no issues so far. But you could use simple DOM if possible.

**Reply by EvanetaI** [reply]
Thank you, guys.
